### PR TITLE
Continue trying after _parse_jwplayer_data fails

### PR DESCRIPTION
**Pull request by moyamo**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This pull request supersedes #12516. Essentially the problem is that `_parse_jwplayer_data` fails when the "JSON" structure it extracts contains variable references. However it seems an extractor further down in generic.py still works.

This pull requests makes it so that if  `_parse_jwplayer_data` fails instead of giving up youtube-dl continues trying different extractors.

### CODE REVIEW

File: youtube_dl/utils.py [hunk 1]
```diff
@@ -748,6 +748,7 @@ def __init__(self, msg, tb=None, expected=False, cause=None, video_id=None):
             msg = video_id + ': ' + msg
         if cause:
             msg += ' (caused by %r)' % cause
+        self.msg_without_bug_report = msg
```
Comment (dstftw) [inline youtube_dl/utils.py]: No.

File: youtube_dl/extractor/generic.py [hunk 2]
```diff
@@ -3119,9 +3119,13 @@ def _real_extract(self, url):
...
             webpage, video_id, transform_source=js_to_json)
         if jwplayer_data:
-            info = self._parse_jwplayer_data(
-                jwplayer_data, video_id, require_title=False, base_url=url)
-            return merge_dicts(info, info_dict)
+            try:
+                info = self._parse_jwplayer_data(
+                    jwplayer_data, video_id, require_title=False, base_url=url)
+                return merge_dicts(info, info_dict)
+            except ExtractorError, e:
```
Comment (dstftw) [inline youtube_dl/extractor/generic.py]: Invalid syntax.

File: youtube_dl/extractor/generic.py [hunk 3]
```diff
@@ -3119,9 +3119,13 @@ def _real_extract(self, url):
...
+                self.to_screen(e.msg_without_bug_report)
```
Comment (dstftw) [inline youtube_dl/extractor/generic.py]: No.

File: youtube_dl/extractor/generic.py [hunk 4]
```diff
@@ -3119,9 +3119,13 @@ def _real_extract(self, url):
...
+                self.to_screen("Trying different extractor")
```
Comment (dstftw) [inline youtube_dl/extractor/generic.py]: No.

Comment (moyamo) [reply youtube_dl/extractor/generic.py]: I'm not sure what's wrong with this message. Should I just remove them.

Comment (moyamo) [reply youtube_dl/extractor/generic.py]: Feedback would be appreciated.

Comment (dstftw) [reply youtube_dl/extractor/generic.py]: There should not be any messages.

**Pull request closed without merging**